### ISSUE TITLE
Compatibility layer for inflecto/flexus users

### ISSUE THREAD

**Issue by abinoam** [dry-rb/dry-inflector#1]
EDIT:

There'll be no compatibilty layer.
For undertanding some of the differences and reasonings of the inflecto/flexus/dry-inflector see: https://github.com/dry-rb/dry-inflector/wiki/From-Inflecto-Flexus-to-Dry::Inflector

---

I'm thinking about something like:

```ruby
require "dry/inflector/compatibility/flexus"
```

Then we fake Flexus to use Dry::Inflector under the hood. Fixing any incompatibility.
At the same time we craft a document about how to migrate from inflecto/flexus to dry-inflecto.

See Ptico/flexus#24

**Comment by jodosha** [comment]
@abinoam Please tell me more. Do you want `dry-inflector` to distribute a file with these contents? https://github.com/Ptico/flexus/pull/24/files#diff-93830fa29d616f7c87903d08b5b1b29a

**Comment by abinoam** [comment]
No. This is just a draft.
When we have craft this "compatibility layer" we may distribute it altogether or, perhaps better, in a separate gem.

**Comment by abinoam** [comment]
I just want something as easy as 

```ruby
require "dry/inflector/compatibility/flexus"
```

Any suggestions? Ideas?
to work out of the box (or almost).

**Comment by solnic** [comment]
Do you really think that a compatibility layer is needed? Moving to dry-inflector means `s/Inflecto/my_inflector_instance/g` in a project after all.

**Comment by abinoam** [comment]
I'll keep searching the "corner use cases".  If I don't find any of them, then you're probably right. We should just update the docs to show how easy would be to migrate.

**Comment by abinoam** [comment]
@solnic I have just found one method that is missing in Dry::Inflector. `Inflecto#foreign_key`
So, just searching and replacing will no work for some people.
If this method is missing **per design**, we shouldn't pollute Dry::Inflector with it.
We just put it on a compatibility layer. If there's much more than this, we should even put them in a separate gem.

I'll have to have a break now. Some work to be done. But I'll keep going on this as soon as possible.

**Comment by abinoam** [comment]
Summarizing.

There's 2 methods missing in `Dry::Inflector`.
`#foreign_id` and `#constantize`.

@jodosha, did you droped these 2 methods intentionally?
I don't know if they are significant.

I already commented that I don't know the difference between `#constantize` and the builtin `Object#const_get`. See: https://github.com/Ptico/flexus/pull/13#issue-270846868

`#foreign_id`... well... it assumes that all foreign ids are following the same standards as Rails. I don't know if this should be in Dry::Inflector. 

What I mean, if you missed them by design, I think I agree with you.

And, now that I see that there's not THAT much of a difference, perhaps @solnic is right and we should just make good documentation so people trying to migrate from `Inflecto` could be confortable with `Dry::Inflector`

**Comment by abinoam** [comment]
@jodosha,

I see that `#humanize` already takes into account the `_id` ending of foreign keys.
```ruby
    def humanize(input)
      result = inflections.humans.apply_to(input)
      result.gsub!(/_id\z/, "")
      result.tr!("_", " ")
      result.capitalize!
      result
    end
```

So, it makes sense to have a `foreign_key` method on inflector. What do you think?

**Comment by abinoam** [comment]
https://github.com/dry-rb/dry-inflector/wiki/From-Inflecto-Flexus-to-Dry::Inflector
☝️ @jodosha @solnic @Ptico @mbj

What do you think of this page?
Feel free to suggest, modify, anything.
As @solnic said, it's so easy to migrate that I think just a good doc will be enough.

**Comment by jodosha** [comment]
@abinoam 
> So, it makes sense to have a foreign_key method on inflector. What do you think?

👍

**Comment by mbj** [comment]
I do not think such a compatbility layer should be required, users that do not wish to have breaking changes (or semantic changes) need to stay on (graveyard) `inflecto` anyway. And the ones who are okay with changes can just directly use `dry-inflector`.

But this is not my project, so this is not really a judgement here.

**Comment by abinoam** [comment]
@mbj your opinions here are really welcome! 👍😄

**Comment by timriley** [comment]
@abinoam I like the wiki page you made. I think something like that is all we need to help people with migration, rather than any special code built into the gem.

My only thought on that wiki page is that it might make sense to split them up into 2:

- Migrating from inflecto
- Migrating from flexus

I have a feeling that 99.9% of the migrations will be from inflecto, given that flexus' lifespan was so short, so removing the flexus-related details from the inflecto doc will make it clearer for the majority of users wanting to migrate.

**Comment by abinoam** [comment]
Hi @timriley thanks for your feedback. I'll try to do the suggestion ASAP! 👍

**Comment by abinoam** [comment]
@timriley, I'll try to clarify why I put it all together inflecto/flexus.

The last inflecto release is from 2013-01-22 (https://rubygems.org/gems/inflecto)
Since then, there's some code available at master branch that were never released as inflecto.
Some people were already using inflecto directly from its github master branch since 2013.
After some discussion, @mbj decided not to release the accumulated code, as it could break legacy applications and libs that depended on inflecto 0.2.0 api.
Then we together decided that forking and releasing it as flexus would be the right thing to do.

So, basically people were **already** using the current flexus code as inflecto master branch. So, it's hard to detach completely.

But, I'll try to make some distinction at the wiki.

cc: @Ptico @mbj

**Comment by abinoam** [comment]
@timriley, just updated the wiki with a better separation between Inflecto/Flexus/Dry::Inflector and a better chronological narrative so people could find their path to migrating to Dry::Inflector.

**Comment by solnic** [comment]
Can we close this? Seems like we agreed that building a compat layer is not worth the effort?

**Comment by mbj** [comment]
> Can we close this? Seems like we agreed that building a compat layer is not worth the effort?

Yeah, just point people from inflecto / flexus to `dry-inflecto` as a compat layer is only increasing everybodies pain. Maintainer time and attention is a limited resource.

People who right now use inflecto and need to update to something maintained likely better switch to `dry-inflector` in one step rather than having to go through a potentially "slightly" semantic varying compat layer.
